### PR TITLE
CLID-66,CLID-81: refines v2 --help

### PR DESCRIPTION
**Pull request by aguidirh**
# Description

This PR changes the --help for v2 (description, usage and examples).

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

```
./bin/oc-mirror --help --v2
```

## Expected Outcome
A different --help output from the main branch

### CODE REVIEW

File: v2/pkg/cli/executor.go [hunk 1]
```diff
@@ -44,22 +44,59 @@ import (
 var (
 	mirrorlongDesc = templates.LongDesc(
 		` 
-		Create and publish user-configured mirrors with a declarative configuration input.
-		used for authenticating to the registries. 
+		Mirror OCP Release, operator catalog and additional images using a declarative configuration file as an input. 
 
-		The podman location for credentials is also supported as a secondary location.
+		The command line uses a declarative configuration file as the input to discover from where to get the container images and copy them to the destination specified in the command line.
```
Comment (lmzuccarelli) [inline v2/pkg/cli/executor.go:49]:
```suggestion
		The command line uses a declarative configuration file as the input to discover where to get the container images and copy them to the destination specified in the command line. 
```

File: v2/pkg/cli/executor.go [hunk 2]
```diff
@@ -44,22 +44,59 @@ import (
...
+		Mirror OCP Release, operator catalog and additional images using a declarative configuration file as an input. 
 
-		The podman location for credentials is also supported as a secondary location.
+		The command line uses a declarative configuration file as the input to discover from where to get the container images and copy them to the destination specified in the command line. 
 
-		1. Destination prefix is docker:// - The current working directory will be used.
-		2. Destination prefix is oci:// - The destination directory specified will be used.
+		There are three workflows available in oc-mirror currently:
+			
+			- mirrorToDisk - pulls the container images from the source specified in the image set configuration and pack them into a tar archive on disk (local directory).
```
Comment (lmzuccarelli) [inline v2/pkg/cli/executor.go:53]:
```suggestion
			- mirrorToDisk - pulls the container images from the source specified in the image set configuration and packs them into a tar archive on disk (local directory).
```

File: v2/pkg/cli/executor.go [hunk 3]
```diff
@@ -44,22 +44,59 @@ import (
...
+			- diskToMirror - copy the containers images from the tar archive to a container registry (--from flag is required on this workflow).
+			- mirrorToDisk - copy the container images from the source specified in the image set configuration to the destination (container registry).
+
+		When specifying the destination on the command line, there are two prefixes available:
+
+			- file://<destination location> - used in mirrorToDisk: local mirror packed into a tar archive.
+			- docker://<destination location> - used in diskToMirror and mirrorToMirror: when the destination is a registry.
+
+		The default podman credentials location ($XDG_RUNTIME_DIR/containers/auth) is used for authenticating to the registries. The docker location for credentials is also supported as a secondary location.
```
Comment (lmzuccarelli) [inline v2/pkg/cli/executor.go:62]: Nice !!!!

File: v2/pkg/cli/executor.go [hunk 4]
```diff
@@ -44,22 +44,59 @@ import (
...
+
+		The name of the directory used by oc-mirror as a workspace defaults to the name working-dir. The location of this directory depends on the following:
```
Comment (lmzuccarelli) [inline v2/pkg/cli/executor.go:64]:
```suggestion
		The name of the directory used by oc-mirror as a workspace defaults to the name 'working-dir'. The location of this directory depends on the following:
```

File: v2/pkg/cli/executor.go [hunk 5]
```diff
@@ -44,22 +44,59 @@ import (
...
+
+			- mirrorToDisk: file://<destination location> 
+			- mirrorToMirror: --workspace file://<destination location>
+			
+			In both cases above the working-dir will be under the folder specified in the <destination location>
+
+		There is also a delete command to delete images in a destination. This command is split in two phases:
```
Comment (lmzuccarelli) [inline v2/pkg/cli/executor.go:71]:
```suggestion
		There is also a delete command to delete images from a remote registry (specified in the cli) . This command is split in two phases:
```

File: v2/pkg/cli/executor.go [hunk 6]
```diff
@@ -44,22 +44,59 @@ import (
...
+
+			- Phase 1: using a delete set configuration as an input, oc-mirror discovers all images that needed to be deleted. These images are included in a delete-images file to be consumed as input in the second phase.
+			- Phase 2: using the file generated in first phase, oc-mirror will delete all manifests specified on this file on the destination specified in the command line. It is up to the container registry to run the garbage collector to clean up all the blobs which are not referenced by a manifest. Deleting only manifests is safer since blobs shared between more than one image are not going to be deleted.
```
Comment (lmzuccarelli) [inline v2/pkg/cli/executor.go:74]: Nice and clear !!!

File: v2/pkg/cli/executor.go (hunk 4 above)
Comment (aguidirh) [reply v2/pkg/cli/executor.go]: Done :white_check_mark:

Thanks for the suggestion.

File: v2/pkg/cli/executor.go (hunk 2 above)
Comment (aguidirh) [reply v2/pkg/cli/executor.go]: Done :white_check_mark:

Thanks for the suggestion.

File: v2/pkg/cli/executor.go (hunk 1 above)
Comment (aguidirh) [reply v2/pkg/cli/executor.go]: Done :white_check_mark:

Thanks for the suggestion.

**Pull request merged**